File: scripts/algorithms/R/Remove Nth Node From End of List/Remove Nth Node From End of List.cpp
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* n1 = new ListNode();
         ListNode *temp=head;
         ListNode *p=head;
        ListNode *q=n1;
        n1->next=head;
        int count;
        while(temp)
        { 
         
            if(n>0){
                n--;
            }
            else if(n==0){
            cout<val;
            p=p->next;
            q=q->next;
            }
               temp=temp->next;
        }
        q->next=p->next;
        delete p;
    return n1->next;
    }
};

File: scripts/algorithms/R/Remove Nth Node From End of List/Remove Nth Node From End of List.java
// Runtime: 1 ms (Top 72.17%) | Memory: 42.2 MB (Top 52.18%)
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp = head;
        int len=0;

        if(head==null || head.next==null)
            return null;

        while(temp!=null){
            temp=temp.next;
            len++;
        }

        if(len==n)
            return head.next;

        int frontlen = len-n-1;

        ListNode first=head.next;
        ListNode second = head;

        int count=0;

        while(first!=null){
            if(count==frontlen){
                second.next=first.next;
                break;
            }else{
                first=first.next;
                second=second.next;
                count++;
            }
        }

        return head;
    }
}